292 create development docker image

m-hemmings opened the pull request:
## Purpose

This PR resolves [#292] by introducing a development-focused Docker build configuration (`Dockerfile.dev`) and corresponding `dev.yml` Compose file for use in github actions to build this automatically. These additions streamline local development and debugging workflows by separating the development environment from the production image.

### Summary of Changes

* **`Dockerfile.dev`**:
  Adds a Dockerfile tailored for development. Key features include:

  * Installs build-essential, debugging tools, and full source context
  * Uses a named user (`lind`) for safer file permissions and UID mapping
  * Retains full source history, tooling, and compiler diagnostics

* **`dev.yml`**:
  Adds a Docker Compose configuration for use as a github action workflow to automatically build it and push to docker hub:

## Linked Issue

Closes [#292]

## Notes for Reviewers

* This is meant to be **non-intrusive**—no changes are required to the main Dockerfile or CI.
* This enables contributors to spin up a consistent development environment without manual setup.
* UID/GID mapping should reduce friction for Linux users mounting source directories.
* If further `.devcontainer` or VS Code-specific config is desired, we can extend this later.

rennergade reviewed (dismissed):
I tested this last night and it looks great to me.

@Yaxuan-w any thoughts?

rennergade reviewed (changes requested):
Small doc change request

m-hemmings commented on docs/getting-started.md:
````diff
@@ -16,6 +16,35 @@ docker pull --platform=linux/amd64 securesystemslab/lind-wasm  # this might take
 docker run --platform=linux/amd64 -it securesystemslab/lind-wasm /bin/bash
 ```
 
+If you would rather use an environment with the source code and tooling, there is a 
````
Great ideas

rennergade replied:
I think everything here should be removed since its in the new file except the last line.

m-hemmings commented:
[6ba1667](https://github.com/Lind-Project/lind-wasm/pull/293/commits/6ba16672509d739e58d21d5b611817e1400ed1da) addresses comment suggestion made by @Yaxuan-w [here](https://github.com/Lind-Project/lind-wasm/issues/292#issuecomment-3078786506)
